landlab/components/detachment_ltd_erosion/generate_erosion_by_depth_slope.py

```diff
@@ -113,44 +117,45 @@ class DepthSlopeProductErosion(Component):
     incision.
     """
 
-    _name = "DetachmentLtdErosion"
-
-    _input_var_names = (
-        "topographic__elevation",
-        "topographic__slope",
-        "surface_water__depth",
-    )
-
-    _output_var_names = ("topographic__elevation",)
-
-    _var_units = {
-        "topographic__elevation": "m",
-        "topographic__slope": "-",
-        "surface_water__depth": "m",
-    }
-
-    _var_mapping = {
-        "topographic__elevation": "node",
-        "topographic__slope": "node",
-        "surface_water__depth": "node",
-    }
-
-    _var_doc = {
-        "topographic__elevation": "Land surface topographic elevation",
-        "topographic__slope": "Slope of the land surface",
-        "surface_water__depth": "Depth of water on the surface",
+    _name = "DepthSlopeProductErosion"
+
+    _info = {
+        "surface_water__depth": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "Depth of water on the surface",
+        },
+        "topographic__elevation": {
+            "dtype": float,
+            "intent": "inout",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "Land surface topographic elevation",
+        },
+        "topographic__slope": {
+            "dtype": float,
+            "intent": "in",
+            "optional": True,
+            "units": "-",
+            "mapping": "node",
+            "doc": "Gradient of surface, on links",
+        },
     }
 
     def __init__(
         self,
         grid,
-        k_e,
+        k_e=0.001,
         fluid_density=1000.0,
-        g=9.81,
+        g=scipy.constants.g,
```

Comment:
I'm going to let @mcflugen respond to this one as I made the `scipy.constants.g` changes in response to his recommendation.

Reply:
Previously, components that used *g* defined it as different values. Some used `scipy.constants.g`, some used 9.81, some used 10 (ok, not really). Just to be consistent, it seems like using the more accurate `scipy.constants.g` (defined as 9.80665) is the way to go. Having it here alerts the user that they are able to change it if they like, although, it is up to the component if it can be spatially variable. Not having it as a keyword argument would really be hiding it. I'm not sure if any components do that any more or not.

A user looking at the documentation for the component will see `g=9.80665` rather than `g=scipy.constants.g`, so they'll know exactly what the default value being used is.

Reply:
That makes sense, given that the actual numerical value will be readily apparent to users.
(by the way, I meant "decimal" not "dimension" ;)